src/storage/rdb.cc

```diff
@@ -680,3 +679,323 @@ Status RDB::LoadRdb(uint32_t db_index, bool overwrite_exist_key) {
+      return {Status::RedisExecErr, s.ToString()};
+    }
+    return SaveSetObject(members);
+  } else if (type == kRedisZSet) {
+    redis::ZSet zset_db(storage_, ns_);
+    std::vector<MemberScore> member_scores;
+    RangeScoreSpec spec;
+    auto s = zset_db.RangeByScore(key, spec, &member_scores, nullptr);
+    if (!s.ok()) {
+      return {Status::RedisExecErr, s.ToString()};
+    }
+    std::sort(member_scores.begin(), member_scores.end(),
+              [](const MemberScore &v1, const MemberScore &v2) { return v1.score > v2.score; });
+    return SaveZSetObject(member_scores);
+  } else if (type == kRedisHash) {
+    redis::Hash hash_db(storage_, ns_);
+    std::vector<FieldValue> field_values;
+    auto s = hash_db.GetAll(key, &field_values);
+    if (!s.ok()) {
+      return {Status::RedisExecErr, s.ToString()};
+    }
+
+    return SaveHashObject(field_values);
+  } else {
+    LOG(WARNING) << "Invalid or Not supported object type: " << type;
+    return {Status::NotOK, "Invalid or Not supported object type"};
+  }
+}
+
+Status RDB::RdbSaveLen(uint64_t len) {
+  unsigned char buf[2];
+  if (len < (1 << 6)) {
+    /* Save a 6 bit len */
+    buf[0] = (len & 0xFF) | (RDB6BitLen << 6);
+    auto status = stream_->Write((const char *)buf, 1);
+    if (!status.IsOK()) {
+      return {Status::RedisExecErr, status.Msg()};
+    }
+    return Status::OK();
+  } else if (len < (1 << 14)) {
+    /* Save a 14 bit len */
+    buf[0] = ((len >> 8) & 0xFF) | (RDB14BitLen << 6);
+    buf[1] = len & 0xFF;
+    auto status = stream_->Write((const char *)buf, 2);
+    if (!status.IsOK()) {
+      return {Status::RedisExecErr, status.Msg()};
+    }
+    return Status::OK();
+  } else if (len <= UINT32_MAX) {
+    /* Save a 32 bit len */
+    buf[0] = RDB32BitLen;
+    auto status = stream_->Write((const char *)buf, 1);
+    if (!status.IsOK()) {
+      return {Status::RedisExecErr, status.Msg()};
+    }
+
+    uint32_t len32 = htonl(len);
+    status = stream_->Write((const char *)(&len32), 4);
+    if (!status.IsOK()) {
+      return {Status::RedisExecErr, status.Msg()};
+    }
+    return Status::OK();
+  } else {
+    /* Save a 64 bit len */
+    buf[0] = RDB64BitLen;
+    auto status = stream_->Write((const char *)buf, 1);
+    if (!status.IsOK()) {
+      return {Status::RedisExecErr, status.Msg()};
+    }
+
+    len = htonu64(len);
+    status = stream_->Write((const char *)(&len), 8);
+    if (!status.IsOK()) {
+      return {Status::RedisExecErr, status.Msg()};
+    }
+    return Status::OK();
+  }
+}
+
+Status RDB::SaveStringObject(const std::string &value) {
+  const size_t len = value.length();
+  int enclen = 0;
+
+  // When the length is less than 11, value may be an integer,
+  // so special encoding is performed.
+  if (len <= 11) {
+    unsigned char buf[5];
+    // convert string to long long
+    std::istringstream iss(value);
+    long long integer_value = 0;
+    if (iss.eof() && !iss.fail()) {
+      iss >> integer_value;
```

Comment:
Could you please use `util::ParseInt` instead?

Reply:
done